src/shared/selinux-util.c

```diff
@@ -30,6 +27,91 @@
 #include "time-util.h"
 
 #if HAVE_SELINUX
+static void *libselinux_dl = NULL;
+
+static DLSYM_PROTOTYPE(context_free) = NULL;
+static DLSYM_PROTOTYPE(context_new) = NULL;
+static DLSYM_PROTOTYPE(context_range_get) = NULL;
+static DLSYM_PROTOTYPE(context_range_set) = NULL;
+static DLSYM_PROTOTYPE(context_str) = NULL;
+DLSYM_PROTOTYPE(avc_open) = NULL;
+DLSYM_PROTOTYPE(fgetfilecon_raw) = NULL;
+DLSYM_PROTOTYPE(fini_selinuxmnt) = NULL;
+DLSYM_PROTOTYPE(freecon) = NULL;
+DLSYM_PROTOTYPE(getcon_raw) = NULL;
+DLSYM_PROTOTYPE(getfilecon_raw) = NULL;
+DLSYM_PROTOTYPE(getpeercon_raw) = NULL;
+DLSYM_PROTOTYPE(getpidcon) = NULL;
+DLSYM_PROTOTYPE(is_selinux_enabled) = NULL;
+DLSYM_PROTOTYPE(security_compute_create_raw) = NULL;
+DLSYM_PROTOTYPE(security_getenforce) = NULL;
+DLSYM_PROTOTYPE(selabel_close) = NULL;
+DLSYM_PROTOTYPE(selabel_lookup_raw) = NULL;
+DLSYM_PROTOTYPE(selabel_open) = NULL;
+DLSYM_PROTOTYPE(selinux_check_access) = NULL;
+DLSYM_PROTOTYPE(selinux_init_load_policy) = NULL;
+DLSYM_PROTOTYPE(selinux_path) = NULL;
+DLSYM_PROTOTYPE(selinux_set_callback) = NULL;
+DLSYM_PROTOTYPE(selinux_status_close) = NULL;
+DLSYM_PROTOTYPE(selinux_status_getenforce) = NULL;
+DLSYM_PROTOTYPE(selinux_status_open) = NULL;
+DLSYM_PROTOTYPE(selinux_status_policyload) = NULL;
+DLSYM_PROTOTYPE(setcon_raw) = NULL;
+DLSYM_PROTOTYPE(setexeccon) = NULL;
+DLSYM_PROTOTYPE(setfilecon) = NULL;
+DLSYM_PROTOTYPE(setfilecon_raw) = NULL;
+DLSYM_PROTOTYPE(setfscreatecon_raw) = NULL;
+DLSYM_PROTOTYPE(setsockcreatecon) = NULL;
+DLSYM_PROTOTYPE(setsockcreatecon_raw) = NULL;
```

Comment:
smilar for set*con, we probably should always use the raw versions, i.e. avoid hidden IPC